From 0cf2785f62030fc2a9ba9d2b64203abe7fb2e3c8 Mon Sep 17 00:00:00 2001
From: Leena Winterrowd <lenhardw@codeaurora.org>
Date: Wed, 7 Jan 2015 19:55:47 -0800
Subject: [PATCH 77/95] stagefright: http: Skip invalid bandwidth measurements

Measurements of 0 bytes downloaded can skew bandwidth estimation.
If 0 bytes or an invalid delay are added as a measurement, do not
add it to the bandwidth history.

Allow bandwidth estimation with a single history entry since the
total delay is now guaranteed to be non-zero for any entry.

CRs-Fixed: 808563
Change-Id: Ib68122302176f892989e37ba26e9848f0f4113d0
---
 media/libstagefright/HTTPBase.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/media/libstagefright/HTTPBase.cpp b/media/libstagefright/HTTPBase.cpp
index 0c2ff15..1593a38 100644
--- a/media/libstagefright/HTTPBase.cpp
+++ b/media/libstagefright/HTTPBase.cpp
@@ -44,6 +44,10 @@ void HTTPBase::addBandwidthMeasurement(
         size_t numBytes, int64_t delayUs) {
     Mutex::Autolock autoLock(mLock);
 
+    if (numBytes == 0 || delayUs <= 0) {
+        return;
+    }
+
     BandwidthEntry entry;
     entry.mDelayUs = delayUs;
     entry.mNumBytes = numBytes;
@@ -75,7 +79,7 @@ void HTTPBase::addBandwidthMeasurement(
 bool HTTPBase::estimateBandwidth(int32_t *bandwidth_bps) {
     Mutex::Autolock autoLock(mLock);
 
-    if (mNumBandwidthHistoryItems < 2) {
+    if (mNumBandwidthHistoryItems < 1) {
         return false;
     }
 
-- 
2.5.0

